src/sbp_fileio.c

```diff
@@ -81,16 +81,16 @@ static void read_cb(u16 sender_id, u8 len, u8 msg[], void* context)
   len += cfs_read(f, buf + len, readlen);
   cfs_close(f);
 
-  sbp_send_msg(SBP_MSG_FILEIO_READ, len, buf);
+  sbp_send_msg(SBP_MSG_FILEIO_READ_RESPONSE, len, buf);
```

Comment:
above here (line 77), we need a check that len + readlen < 256, otherwise the you could easily overrun buffer[256].  I don't think we can trust the sender of the message took into account that readlen needs to be correct.